app/services/service_listeners/publishing_api_pusher.rb

```diff
@@ -0,0 +1,36 @@
+module ServiceListeners
+  class PublishingApiPusher
+    attr_reader :edition
+
+    def initialize(edition)
+      @edition = edition
+    end
+
+    def push(event:, options: {})
+      case event
+      when "force_publish", "publish"
+        api.publish_async(edition)
+      when "update_draft"
+        api.save_draft_async(edition)
+      when "update_draft_translation"
```

Comment:
We should have unit test coverage for all the cases here. Currently only the first three are covered.